compass/landice/tests/slm/circ_icesheet/setup_mesh.py

```diff
@@ -0,0 +1,409 @@
+    put_origin_on_a_cell = section.getboolean('put_origin_on_a_cell')
+
+    # Open the file, get needed dimensions
+    gridfile = NetCDFFile(filename, 'r+')
+    nVertLevels = len(gridfile.dimensions['nVertLevels'])
+    # Get variables
+    xCell = gridfile.variables['xCell']
+    yCell = gridfile.variables['yCell']
+    xEdge = gridfile.variables['xEdge']
+    yEdge = gridfile.variables['yEdge']
+    xVertex = gridfile.variables['xVertex']
+    yVertex = gridfile.variables['yVertex']
+    thickness = gridfile.variables['thickness']
+    bedTopography = gridfile.variables['bedTopography']
+    layerThicknessFractions = gridfile.variables['layerThicknessFractions']
+
+    # Find center of domain
+    x0 = xCell[:].min() + 0.5 * (xCell[:].max() - xCell[:].min())
+    y0 = yCell[:].min() + 0.5 * (yCell[:].max() - yCell[:].min())
+    # Calculate distance of each cell center from dome center
+    r = ((xCell[:] - x0) ** 2 + (yCell[:] - y0) ** 2) ** 0.5
+
+    if put_origin_on_a_cell:
+        # Center the ice in the center of the cell that is closest to the
+        # center of the domain.
+        centerCellIndex = np.abs(r[:]).argmin()
+        xShift = -1.0 * xCell[centerCellIndex]
+        yShift = -1.0 * yCell[centerCellIndex]
+        xCell[:] = xCell[:] + xShift
+        yCell[:] = yCell[:] + yShift
+        xEdge[:] = xEdge[:] + xShift
+        yEdge[:] = yEdge[:] + yShift
+        xVertex[:] = xVertex[:] + xShift
+        yVertex[:] = yVertex[:] + yShift
+        # Now update origin location and distance array
+        x0 = 0.0
+        y0 = 0.0
+        r = ((xCell[:] - x0) ** 2 + (yCell[:] - y0) ** 2) ** 0.5
+
+    # Assign variable values for the circular ice sheet
+    # Set default value for non-circular cells
+    thickness[:] = 0.0
+    # Calculate the dome thickness for cells within the desired radius
+    # (thickness will be NaN otherwise)
+    thickness_field = thickness[0, :]
+
+    if ice_type == 'cylinder':
+        logger.info('cylinder ice type is chosen')
+        thickness_field[r < r0] = h0
+    elif ice_type == 'dome-cism':
+        thickness_field[r < r0] = h0 * (1.0 - (r[r < r0] / r0) ** 2) ** 0.5
+    elif ice_type == 'dome-halfar':
+        thickness_field[r < r0] = h0 * (
+            1.0 - (r[r < r0] / r0) ** (4.0 / 3.0)) ** (3.0 / 7.0)
+    else:
+        raise ValueError('Unexpected ice_type: {}'.format(ice_type))
+    thickness[0, :] = thickness_field
+
+    # flat bed at sea level
+    bedTopography[:] = 0.0
+    if set_topo_elev:
+        # this line will make a small shelf:
+        bedTopography[:] = topo_elev
+
+    # Setup layerThicknessFractions
+    layerThicknessFractions[:] = 1.0 / nVertLevels
+
+    gridfile.close()
+
+    logger.info('Successfully added circular initial conditions to: {}'.format(
+        filename))
+
+
+def _create_smb_forcing_file(config, logger, mali_mesh_file, filename):
+    """
+    Create surface mass balance forcing file for circular ice sheet
```

Comment:
You or I should add notes here (and maybe also docs) about why we are imposing radial retreat with SMB.